sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobLogOptions.java

```diff
@@ -0,0 +1,163 @@
+        "x-ms-request-id", "x-ms-return-client-request-id", "x-ms-version", "Accept-Ranges", "Content-Disposition",
+        "Content-Encoding", "Content-Language", "Content-MD5", "Content-Range", "ETag", "Last-Modified", "Server",
+        "Vary", "x-ms-content-crc64", "x-ms-copy-action", "x-ms-copy-completion-time", "x-ms-copy-id",
+        "x-ms-copy-progress", "x-ms-copy-status", "x-ms-has-immutability-policy", "x-ms-has-legal-hold",
+        "x-ms-lease-state", "x-ms-lease-status", "x-ms-range", "x-ms-request-server-encrypted",
+        "x-ms-server-encrypted", "x-ms-snapshot", "x-ms-source-range", "If-Match", "If-Modified-Since",
+        "If-None-Match", "If-Unmodified-Since", "x-ms-access-tier", "x-ms-access-tier-change-time",
+        "x-ms-access-tier-inferred", "x-ms-account-kind", "x-ms-archive-status", "x-ms-blob-append-offset",
+        "x-ms-blob-cache-control", "x-ms-blob-committed-block-count", "x-ms-blob-condition-appendpos",
+        "x-ms-blob-condition-maxsize", "x-ms-blob-content-disposition", "x-ms-blob-content-encoding",
+        "x-ms-blob-content-language", "x-ms-blob-content-length", "x-ms-blob-content-md5", "x-ms-blob-content-type",
+        "x-ms-blob-public-access", "x-ms-blob-sequence-number", "x-ms-blob-type", "x-ms-copy-destination-snapshot",
+        "x-ms-creation-time", "x-ms-default-encryption-scope", "x-ms-delete-snapshots",
+        "x-ms-delete-type-permanent", "x-ms-deny-encryption-scope-override", "x-ms-encryption-algorithm",
+        "x-ms-if-sequence-number-eq", "x-ms-if-sequence-number-le", "x-ms-if-sequence-number-lt",
+        "x-ms-incremental-copy", "x-ms-lease-action", "x-ms-lease-break-period", "x-ms-lease-duration",
+        "x-ms-lease-id", "x-ms-lease-time", "x-ms-page-write", "x-ms-proposed-lease-id",
+        "x-ms-range-get-content-md5", "x-ms-rehydrate-priority", "x-ms-sequence-number-action", "x-ms-sku-name",
+        "x-ms-source-content-md5", "x-ms-source-if-match", "x-ms-source-if-modified-since",
+        "x-ms-source-if-none-match", "x-ms-source-if-unmodified-since", "x-ms-tag-count",
+        "x-ms-encryption-key-sha256")
+        .collect(Collectors.toCollection(HashSet::new));
+
+    private static final Set<String> BLOB_QUERY_PARAMETERS = Stream.of(
+        "comp", "maxresults", "rscc", "rscd", "rsce", "rscl", "rsct", "se", "si", "sip", "sp", "spr", "sr", "srt",
+        "ss", "st", "sv", "include", "marker", "prefix", "copyid", "restype", "blockid", "blocklisttype",
+        "delimiter", "prevsnapshot", "ske", "skoid", "sks", "skt", "sktid", "skv", "snapshot")
+        .collect(Collectors.toCollection(HashSet::new));
+
+    private Set<String> allowedHeaderNames;
+    private Set<String> allowedQueryParamNames;
+
+    /**
+     * Constructor for combining core and Storage Blob allowed whitelist headers and queries.
+     */
+    public BlobLogOptions() {
+        this.allowedHeaderNames = getHeaderSet();
```

Comment:
Are we losing the default headers and query parameters the base HttpLogOptions is adding? I'm still not certain on having subclasses to handle this.

@JonathanGiles Input on subclasses?

Reply:
I am not a fan of creating subclasses whenever we want to have a different set of default whitelisted headers. I would like to see a better approach than this.

Reply:
I'm curious why we needed to subclass this rather than doing something like:
```java
options.addAllowedHeader(… headers);
options.addAllowedQueryParameter(… parameters);
```

Two notes about the HttpLogOptions:
* Do we need to suffix these methods with "Names"? I feel like it's unnecessarily verbose.
* I wish HttpLogOptions.addAllowedHeaderName(), I wish took an Iterable or varargs if we wanted to add more than one header.

Reply:
I was thinking to force customer to have the storage headers when do the setter.
Current design is to add a new default getter in builder helper.